fe/fe-core/src/main/java/com/starrocks/service/FrontendOptions.java

```diff
@@ -127,37 +126,81 @@ public static void init(String[] args) throws UnknownHostException {
             System.exit(-1);
         }
         fileStoredHostType = prop.getProperty(HOST_TYPE, null);
+        // Check if the ROLE file has property 'hostType'
+        // If it not has property 'hostType', start with IP
+        // If it has property 'hostType' & hostType = IP, start with IP
         if (null == fileStoredHostType || fileStoredHostType.equals(HostType.IP.toString())) {
             initAddrUseIp(hosts);
             return;
         }
+        // If it has property 'hostType' & hostType = FQDN, start with FQDN
         initAddrUseFqdn(hosts);
     }
 
     @VisibleForTesting
-    static void initAddrUseFqdn(List<InetAddress> hosts) throws UnknownHostException {
+    static void initAddrUseFqdn(List<InetAddress> hosts) {
         useFqdn = true;
-        InetAddress uncheckedLocalAddr = InetAddress.getLocalHost();
-        if (null == uncheckedLocalAddr) {
-            LOG.error("get a null localhost when start fe use fqdn");
+
+        // Try to get FQDN from host
+        Process hostname = null;
+        try {
+            hostname = Runtime.getRuntime().exec("hostname -f");
+        } catch (IOException e) {
+            LOG.error("Got a IOException when try to get FQDN, message:{}", e.getMessage());
+            System.exit(-1);
```

Comment:
I suggest you raise an exception instead of exit brutally. It's also easier to test.

Reply:
I will push a new PR resolve this problem